### PR TITLE
Feature/tca 399/adjusted remaining time

### PR DESCRIPTION
**Pull request by augustas**
Timer adjustments are included when calculating remaining time for proctor delivery execution list
 
Related to : https://oat-sa.atlassian.net/browse/TCA-399
 
During timer adjustment remaining time cache is updated in monitoring cache. In case extra time would be set as well (f.e. current `extended_time` LTI parameter is using it), then extra time recalculation is also including timer adjustment.
 
#### How to test
 
Manual testing can be performed in playground environment created for this feature or local environment with appropriate branches of `taoTests`, `taoQtiTest`, `taoProctoring`. The configuration option `DeliveryExecutionStateService::OPTION_TIME_HANDLING` should be set to `DeliveryExecutionStateService::TIME_HANDLING_TIMER_ADJUSTMENT`.

- As a test taker launch a proctored test with time limits
- As a proctor authorize the test to start, when the test is in progress pause it
- As a proctor refresh the delivery execution list
- As a proctor click on Change time button, specify how many minutes should be added/subtracted and click OK. Remaining time in the list should correspond to the adjustment.
  
#### Dependencies
   
Companion PR :
 - [ ] https://github.com/oat-sa/extension-tao-testqti/pull/1750
 - [ ] https://github.com/oat-sa/extension-tao-act/pull/1352

### CODE REVIEW

File: model/execution/DeliveryExecutionManagerService.php [hunk 1]
```diff
@@ -209,7 +219,7 @@ public function setExtraTime($deliveryExecutions, $extraTime = 0, $extendedTime
                     $durationStore[$testDefinition->getIdentifier()] = new QtiDuration("PT${newSeconds}S");
 
                     $testSessionService->persist($testSession);
-                    $maxTime = $this->getPartTimeLimits($testSession);
+                    $maxTime = $this->getPartTimeLimits($testSession, $testDefinition);
```
[the remark this answers is not in the document]
Comment (augustas) [reply model/execution/DeliveryExecutionManagerService.php:222]: Before this line did not work at all, I fixed it according to what I thought was original developer intention.

File: model/execution/DeliveryExecutionManagerService.php [hunk 2]
```diff
@@ -81,14 +81,24 @@ public function getDeliveryTimer($deliveryExecution)
     }
 
     /**
+     * @param TestSession $testSession
      * @param $part
      * @return int|null
      */
-    protected function getPartTimeLimits($part)
+    protected function getPartTimeLimits($testSession, $part)
```
[the remark this answers is not in the document]
Comment (augustas) [reply model/execution/DeliveryExecutionManagerService.php:88]: `$testSession` might be undefined, when this method is called from an LTI launch with `extended_time` parameter

File: model/execution/DeliveryExecutionManagerService.php [hunk 3]
```diff
@@ -81,14 +81,24 @@ public function getDeliveryTimer($deliveryExecution)
...
     {
         $timeLimits = $part->getTimeLimits();
-        if ($timeLimits && $timeLimits->hasMaxTime()) {
-            return $timeLimits->getMaxTime()->getSeconds(true);
+        if ($timeLimits && ($maxTime = $timeLimits->getMaxTime()) !== null) {
```
[the remark this answers is not in the document]
Comment (augustas) [reply model/execution/DeliveryExecutionManagerService.php:91]: I've extracted the adjustment application to `TimerAdjustmentServiceInterface::getAdjustedMaxTime(...)`

**Comment by SashaTAO** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Code is self documented

**Pull request merged**